### PR TITLE
Localized Lookup Table Generation

### PR DESCRIPTION
**Pull request by NathanaelMangold**
I added the ability to get localized lookup tables for different language clients.

So using e.g.
`cli-main.py gen_lookup_tables --lang deDE`
create the on for the German client.

Also I added UTF-8 encoding to the files which is really important for other languages than English to get the Umlauts working in game.

The only thing left to discuss is the way the localized query is done. Should we keep it as a separate SQLs (with duplication of the code) or should we somehow merge the two SQLs together?

### CODE REVIEW

File: tts_cli/tts_utils.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def preprocess_dataframe(self, df):
         df['gender'] = df['DisplaySexID'].map(GENDER_DICT)
         df['voice_name'] = df['race'] + '-' + df['gender']
 
-        df['templateText_race_gender'] = df['text'] + df['race'] + df['gender']
+        df['templateText_race_gender'] = df['original_text'] + df['race'] + df['gender']
```
Comment (NathanaelMangold) [inline tts_cli/tts_utils.py:141]: Here I had to change the function to always use the English text so the hash values still matches.

File: tts_cli/utils.py [hunk 2]
```diff
@@ -13,4 +13,27 @@ def get_last_n_words(text, n):
 def replace_dollar_bs_with_space(text):
     pattern = r'(\$[Bb])+'
     result = re.sub(pattern, ' ', text)
-    return result
\ No newline at end of file
+    return result
+
+def language_code_to_language_number(local_code : str) -> int:
```
Comment (NathanaelMangold) [inline tts_cli/utils.py:18]: All languages from the MaNGOS DB

**Comment by NathanaelMangold** [comment]
@xvwyh thanks for the additions.

I compared the generated files and they are identical. Also checked in the German client for 3.3.5a and it worked. So from my end its looking good.

**Review by mrthinger — APPROVED** [review]

**Comment by mrthinger** [comment]
looks good. Thanks

**Pull request merged**